### DISCUSSION TITLE
Can I drop indexes during `createMany`?

### DISCUSSION THREAD

**Discussion by rmx90210** [prisma/prisma#23075] [Q&A]
### Question

Hello! I need to insert several million rows into a Postgres DB and I was planning on using `createMany`. If I were doing this with raw SQL, I'd typically drop indexes first and then rebuild them after the bulk-insert is finished. Is it possible to do this with Prisma? Is there another technique that's recommended?

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment]
Hi @rbrtmrtn 👋 

Thank you for raising this question.

Prisma Client does not have support for dropping and rebuilding index during a bulk insert operation with `createMany`. As a workaround, you can:

1. Create a draft migration using `--create-only` flag.
2. Modify the generated SQL file to drop the index.
3. Apply the modified SQL by running: `npx prisma migrate dev`
4. Create the bulk records using the `createMany` query.
5. Create another draft migration and modify the SQL file to add the index.
6. Apply the modified SQL by running the `npx prisma migrate dev` command

You can also create a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=_) so our engineers can take a look.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
